### PR TITLE
Add VSPS Transformation for ray tracing shaders.

### PR DESCRIPTION
**Pull request by Darkstep79**
- New library process for VSPS Transform.
- Patch the PS storeoutput after scalarizing as this was causing issues with the payload.
- Fix snorm VSPS transform.
- Use dx.op.bitcastI16toF16 instead of LLVM bitcast.
- Add external DXIL.dll validation.
- Remove bitcast of min precision types.
- Fixes for name/space VSPS splitting.
- Added library space/name adjusting.
- Added globals for SRVs.

### CODE REVIEW

**Comment by antiagainst** [comment]
This PR breaks non-Windows platforms. I fixed that with 948de62.

File: lib/Bitcode/Writer/ValueEnumerator.cpp [hunk 1]
```diff
@@ -11,6 +11,9 @@
 //
 //===----------------------------------------------------------------------===//
 
+// jim: TEMP
```
[the remark this answers is not in the document]
Comment (Darkstep79) [reply lib/Bitcode/Writer/ValueEnumerator.cpp]: Yes, will remove this temporary code.

File: lib/Bitcode/Writer/ValueEnumerator.h [hunk 2]
```diff
@@ -128,6 +128,12 @@ class ValueEnumerator {
 
   unsigned getTypeID(Type *T) const {
     TypeMapType::const_iterator I = TypeMap.find(T);
+    // jim: TEMP
```
[the remark this answers is not in the document]
Comment (Darkstep79) [reply lib/Bitcode/Writer/ValueEnumerator.h]: Yes, will remove this temporary code.

File: lib/HLSL/DxilOperations.cpp [hunk 3]
```diff
@@ -1078,6 +1078,16 @@ bool OP::GetOpCodeClass(const Function *F, OP::OpCodeClass &opClass) {
   auto iter = m_FunctionToOpClass.find(F);
   if (iter == m_FunctionToOpClass.end()) {
     // When no user, cannot get opcode.
+    // jim: TEMP
```
[the remark this answers is not in the document]
Comment (Darkstep79) [reply lib/HLSL/DxilOperations.cpp]: Yes, will remove this temporary code.

**Comment by Darkstep79** [comment]
Closing, no longer maintained.

**Pull request closed without merging**